lib/iris/cube.py

```diff
@@ -210,6 +214,54 @@ def __getslice__(self, start, stop):
         result = CubeList(result)
         return result
 
+    def __iadd__(self, other_cubes):
+        """
+        Add a sequence of cubes to the cubelist in place.
+        """
+        super(CubeList, self).__iadd__(CubeList(other_cubes))
+
+    def __setitem__(self, key, cube_or_sequence):
+        """Set self[key] to cube or sequence of cubes"""
+        if isinstance(key, int):
+            # should have single cube.
+            self._assert_is_cube(cube_or_sequence)
+        else:
+            # key is a slice (or exception will come from list method).
+            cube_or_sequence = CubeList(cube_or_sequence)
+
+        super(CubeList, self).__setitem__(key, cube_or_sequence)
+
+    #  __setslice__ is only required for python2.7 compatibility.
```

Comment:
This bit can go 😄

Reply:
Sorry I messed up my branch + re-created.
Hence all-new commits !

Reply:
Now done, I think
https://github.com/SciTools/iris/pull/4767/commits/69f892025de578be9002e919fa26b6d95d9168e2